pkg/cluster/internal/create/kubeadm-init.go

```diff
@@ -103,11 +104,28 @@ func runKubeadmInit(ec *execContext, configNode *NodeReplica) error {
 
 	// install the CNI network plugin
 	// TODO(bentheelder): support other overlay networks
-	if err := node.Command(
-		"/bin/sh", "-c",
-		`kubectl apply --kubeconfig=/etc/kubernetes/admin.conf -f "https://cloud.weave.works/k8s/net?k8s-version=$(kubectl version --kubeconfig=/etc/kubernetes/admin.conf | base64 | tr -d '\n')"`,
-	).Run(); err != nil {
-		return errors.Wrap(err, "failed to apply overlay network")
+	// first probe for a pre-installed manifest
+	haveDefaultCNIManifest := true
+	if err := node.Command("test", "-f", "/kind/manifests/default-cni.yaml").Run(); err != nil {
+		haveDefaultCNIManifest = false
+	}
+	if haveDefaultCNIManifest {
+		// we found the default manifest, install that
+		// the images should already be loaded along with kubernetes
+		if err := node.Command(
+			"kubectl", "create", "--kubeconfig=/etc/kubernetes/admin.conf",
+			"-f", "/kind/manifests/default-cni.yaml",
+		).Run(); err != nil {
+			return errors.Wrap(err, "failed to apply overlay network")
+		}
+	} else {
```

Comment:
given now the images are preloaded and manifest is written on the node, shouldn't we exit with an error instead of going for the download from web method as a fallback?

Reply:
the command below does download from the web?
kind gracefully degrades in the presence of older images without this to the behavior expected for those images  

this is in line with our principles of:
- graceful degredation
- avoiding breaking changes
- avoiding assumptions